pkg/build/util/generate_test.go

```diff
@@ -352,3 +532,22 @@ func mockBuildConfig(source api.BuildSource, strategy api.BuildStrategy, output
 		},
 	}
 }
+
+type mockImageRepositoryNamespaceGetter struct {
+	namespace string
+	name      string
+}
+
+func (m *mockImageRepositoryNamespaceGetter) GetByNamespace(namespace, name string) (*imageapi.ImageRepository, error) {
+	if m.namespace == namespace && m.name == name {
+		return &imageapi.ImageRepository{
+			ObjectMeta: kapi.ObjectMeta{Name: imageRepoName},
+			Status: imageapi.ImageRepositoryStatus{
+				DockerImageRepository: originalImage,
+			},
+			Tags: map[string]string{tagName: newTag},
+		}, nil
+	} else {
```

Comment:
----- Original Message -----
> > +
> > +type mockImageRepositoryNamespaceGetter struct {
> > +	namespace string
> > +	name      string
> > +}
> > +
> > +func (m *mockImageRepositoryNamespaceGetter) GetByNamespace(namespace,
> > name string) (*imageapi.ImageRepository, error) {
> > +	if m.namespace == namespace && m.name == name {
> > +		return &imageapi.ImageRepository{
> > +			ObjectMeta: kapi.ObjectMeta{Name: imageRepoName},
> > +			Status: imageapi.ImageRepositoryStatus{
> > +				DockerImageRepository: originalImage,
> > +			},
> > +			Tags: map[string]string{tagName: newTag},
> > +		}, nil
> > +	} else {
> 
> the current form is more readable than the inverted logic, imho.  I would
> argue that fewer negations in boolean logic is more valuable than fewer
> indentations in code blocks, though i'm sure there are extreme examples for
> either case.

`m.namespace != namespace || m.name != name` doesn't seem like a difficult condition to parse, but I agree this is a borderline case of enforcing the inversion.  The key decider I would argue is that the return at the bottom of the function is the "normal case", while every other case is exceptional or short circuit.  In that context the value of the last return being success is more valuable than the slightly harder boolean.

> 
> also i dislike the practice of not being explicit in having an else block.
> it means the future reader needs to have noticed there is a return in the
> if-block to understand whether or not the non-elsed logic will execute.
> it's practically a goto at that point.  I can see cases where the
> additional else block could reduce readability, but this doesn't seem like
> one of those.

This is our convention - it's better to follow it than to argue it.  The else adds nothing, most if blocks in Go are very clearly exit early functions.  I don't think noticing return is an overly onerous requirement either.